packages/core/src/dialects/abstract/query-generator-typescript.ts

```diff
@@ -813,4 +813,10 @@ Only named replacements (:name) are allowed in literal() because we cannot guara
   versionQuery(): string {
     throw new Error(`${this.dialect.name} did not implement versionQuery`);
   }
+
+  tableExistsQuery(tableName: TableNameOrModel): string {
```

Comment:
It's unfortunate that we can't run integration tests on snowflake, since I think the TABLE_TYPE for snowflake should be `TABLE`. See what is mentioned under the kind column here; https://docs.snowflake.com/en/sql-reference/sql/show-tables#output
But `BASE TABLE` might be the proper usage in this, I couldn't find anything about the possibly table types on the snowflake docs so we need to test it to be sure.
So this comment is more for your information, you don't need to change anything

We've already noticed that the Snowflake implementation sometimes just copies mysql without actually being correct.

Reply:
Thanks for the heads up, I'll take notice of this for future PRs.

Looking at https://www.snowflake.com/blog/using-snowflake-information-schema/, it appears that snowflake does use `BASE TABLE` so this should be fine.

Reply:
You're right. I didn't find that resource since it's in the picture